### PR TITLE
tech spec for monorepo POC (GSI-659)

### PR DESCRIPTION
**Pull request by KerstenBreuer**
See document.

### CODE REVIEW

File: 51-ethiopian-wolf/technical_specification.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+- Which types of tests are required in the mono-repo?
+    - unit tests (per service)
+    - service-wise API tests (conformance tests)
+    - inter-service integration tests
+        - Behavioral
+        - contract-testing
+        - black and/or white box
+- How to ensure that microservice boundraries are respected?
+    - import analysis?
+    - flag PRs that add changes to multiple repositories
```
Comment (TheByronHimes) [inline 51-ethiopian-wolf/technical_specification.md:45]:
```suggestion
    - flag PRs that add changes to multiple services within the monorepo
```

**Review by TheByronHimes — APPROVED** [review]
Looks like a good list of concerns - I was unable to come up with anything substantial to add at the moment. I think the most critical areas to iron out first are the configuration management and versioning/release procedure. Less critical to the services themselves but nice for us would be to determine a good branching strategy early on as well. Taking the wrong approach there might land us in a nice bird's nest of merge conflicts. The other items have less of a long-lasting impact, I think.

**Comment by KerstenBreuer** [comment]
Thanks @TheByronHimes, I fully agree and added the priorities you mentioned to the document.

File: 51-ethiopian-wolf/technical_specification.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    - https://medium.com/@davidsmithtech/dry-and-efficient-python-mono-repos-with-code-example-bc7ee8292e9d
+    - https://betterprogramming.pub/the-pros-and-cons-monorepos-explained-f86c998392e1
+
+
+## Additional Implementation Details:
+
+Moving the existing file services into a monorepo might serve as a good proof of
+concept as these services are fully evolved and understood.
+
+Comming up with a strategy for life cycle management (versioning, releasing, branching)
```
Comment (TheByronHimes) [inline 51-ethiopian-wolf/technical_specification.md:88]:
```suggestion
Coming up with a strategy for life cycle management (versioning, releasing, branching)
```

File: 51-ethiopian-wolf/technical_specification.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+has highest priority and should servce a long-lasting solution. The other aspects are
```
Comment (TheByronHimes) [inline 51-ethiopian-wolf/technical_specification.md:89]:
```suggestion
has highest priority and should service a long-lasting solution. The other aspects are
```

**Review by TheByronHimes — DISMISSED** [review]

**Review by TheByronHimes — APPROVED** [review]

**Pull request merged**